### DISCUSSION TITLE
Issue with BrowserStack when using browser.reloadSession() - there is always an extra test being executed

### DISCUSSION THREAD

**Discussion by jadersonts** [webdriverio/webdriverio#13799] [Q&A] [answered]
## Description

In order to report every test (therefore: `it()` block) in BrowserStack, I tried to use the following snippet in our `wdio.conf` file:

```
    afterTest: async function(test, context, { error, result, duration, passed, retries }) {
         await browser.reloadSession()
    } 
```
Using as example a suite called "Login" (therefore: `describe()` block), having 2 tests: `Login as Coach` and `Login as Player`, we have the results reported, separated and properly recorded in BrowserStack.

However, out of nowhere a 3rd test starts to run in BrowserStack, and it seems to start and end once the results from previous (and official tests) got updated. You can watch below:

https://github.com/user-attachments/assets/b48a10bf-a45d-4f84-9888-84c4104f6fdd

I would like to know how to prevent this "extra test". Is there anything missing here? Or BrowserStack known issue?

Appreciate your help!

**Comment by christian-bromann** [comment]
> I would like to know how to prevent this "extra test". Is there anything missing here? Or BrowserStack known issue?

I would not recommend to call `reloadSession` after every test since re-creating a new browser session takes very long. You can avoid the extra session by not calling `reloadSession` after the last test did run.

**Reply by jadersonts** [reply]
I understand `reloadSession` will take longer but client wants less and less number of `spec` files, and create the reports from each `it` block. Your suggestion by not calling `reloadSession` after the last test did run sounds pretty good. However I am struggling to find a way to check beforehand what is the "last test". For example, I expected to work around `suite` object in `beforeSuite` hook, but there is no useful key/value. Other than that, creating a `countItBlocks.js`, but it seems too radical. Do you have any suggestion in between? Thanks for the support!

**Reply by christian-bromann** [reply]
Unfortunately I don't.

**Comment by jadersonts** [comment, marked as answer]
I created a solution - in `wdio.conf` file I added a `countItBlocksInFile` that will count how many `it` blocks we have in a `spec` file
```
const countItBlocksInFile = (filePath) => {
    const content = fs.readFileSync(filePath, 'utf-8');
    const matches = content.match(/it\(\s*['"`](.*?)['"`]\s*,/g);
    return matches ? matches.length : 0;
}
```
Then I use it in `beforeSuite` hook. At the same time, I added a `counter++` in `beforeTest` hook. So in order to not call `reloadSession` after the last test:
```
afterTest: async function() {
    if(runCounter < numberOfTests){
        await browser.reloadSession()
    }
}
```
Reports are working fine in BrowserStack

**Reply by DhanalakshmiRamasamy** [reply]
@jadersonts - Are you not encountering Session error when you are using Cloud services ? Its because the session reload takes time and the cloud runners are still referring to older session meanwhile.
